magma/digital.py

```diff
@@ -126,6 +127,12 @@ def qualify(cls, direction):
     def __eq__(cls, rhs):
         return cls is rhs
 
+    def is_wireable(cls, rhs):
```

Comment:
@leonardt should we doing these checks on the undirected form of `cls` and `magma_type(rhs)`?

Reply:
The purpose of `is_wireable` is for `Bit` `ite`, `t_type is not f_type`.
https://github.com/phanrahan/magma/blob/687c1aa1174a52f78fdefaab26896b89e5029e7a/magma/bit.py#L149
The problem was, for example, when comparing `Tuple[Bit]` vs `Tuple[MyProtocolBit]`. So the check is supposed to work for nested Array/Tuple/Product.

Reply:
Got it. In that case we should use the undirected version right? The concern is if `In(Bit)` is wireable to `Out(Bit)` or not.

Reply:
maybe `is_wireable` is not a good name. It's `__eq__` for `ite`, only (so, I named it `ite_eq` before.)

Reply:
Is it only used for `ite` right now? I think we should have a free function implementing this logic locally where it is used (we can move it in the future if it gets used elsewhere). Just concerned about attaching this specific of logic to the types.

Reply:
scratch that, because it gets called in bit.py which would then have dependency on Array/Tuple types, which we don't want. Member functions are fine.